pkg/frontend/querymiddleware/shard_by_series_base.go

```diff
@@ -27,6 +27,12 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+const (
+	// ErrMergingPartialResponseStr is the error message used when merging partial responses from sharded queries fails.
+	// This is used by the recommendations service (backend-enterprise repo) to determine whether to retry requests.
```

Comment:
I don't think we should be inspecting error messages to determine if requests should be retried. The query-frontend sets reasonable HTTP status codes and the APIErrors returned set a `type` that can be used to determine if something is transient or not.